modules/perception/camera/app/cipv_camera.cc

```diff
@@ -142,10 +152,10 @@ bool Cipv::MakeVirtualLane(const LaneLineSimple &ref_lane_line,
   virtual_lane_line->line_point.clear();
   if (b_image_based_cipv_ == false) {
     for (uint32_t i = 0; i < ref_lane_line.line_point.size(); ++i) {
-      Eigen::Vector2f virtual_line_point(
+      Point2Df virtual_line_point(
           ref_lane_line.line_point[i](0),
           ref_lane_line.line_point[i](1) + offset_distance);
-      virtual_lane_line->line_point.push_back(virtual_line_point);
+      virtual_lane_line->line_point.emplace_back(virtual_line_point);
```

Comment:
1. Use `push_back` when the object is already constructed.
2. Or use `emplace_back(x, y)` to construct in place which is more efficient.

Reply:
I prefer emplace_back to push_back. Eventually I plan to remove all dynamic data allocations and replace them with fixed_vector or circular_vector